### PR TITLE
Android: Only use touch-sound when view is supposed to receive touches

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-16080

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 1]
```diff
@@ -1720,11 +1720,11 @@ protected void registerForTouch(final View touchable)
 			return;
 		}
 
+		boolean enabled = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_TOUCH_ENABLED), true);
```
Comment (garymathews) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: Should keep this inside

```Java
if (proxy.hasProperty(TiC.PROPERTY_TOUCH_ENABLED)) {
```

Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: restored

**Comment by fmerzadyan** [comment]
Sound effects are still present after change. `enabled` is set to false initially but changes to true which is probably why sound is still enabled.

**Comment by hansemannn** [comment]
This solution will retain the default behavior. Do you have a better idea to solve it more dynamically? The PR was taken from [this commit](https://github.com/gstreetmedia/titanium_mobile/commit/86839826cad0c4e4eeb8f491c3922004161b9545), so we might want to restore the `else` case again, which should disable it by default. Thoughts?

**Comment by hansemannn** [comment]
@fmerzadyan Changed it to disable by default. But I can see that this could cause problems for buttons / list items that do not have the "touchEnabled" property defined manually. Or do we have default values set for this?

**Comment by antw12** [comment]
Hi Hans, I had a look at this ticket and found a potential solution... 

`protected void registerForTouch(final View touchable)
	{
		if (touchable == null) {
			return;
		}

		if (proxy.hasProperty(TiC.PROPERTY_TOUCH_ENABLED)) {
			boolean enabled = TiConvert.toBoolean(proxy.getProperty("soundEffectsEnabled"), true);
			touchable.setSoundEffectsEnabled(enabled);
			touchable.setEnabled(enabled);
		}
		
		registerTouchEvents(touchable);`

Seems that because of the boolean conversion was using the constant PROPERTY_TOUCH_ENABLED which is always true, i think is the reason why it always had sounds 
please let me know if you want me to do anything. thanks

**Comment by hansemannn** [comment]
touchEnabled is not always true, so the latest solution would handle both cases. @garymathews thoughts?

**Comment by garymathews** [comment]
@hansemannn @antw12 I think we should implement a separate property for sound (e.g `PROPERTY_TOUCH_SOUND_ENABLED`) that's independent to `PROPERTY_TOUCH_ENABLED`.

**Pull request closed without merging**